### ISSUE TITLE
terraform plan fails if we have more than 3 instances in one aws_instances.tf.json file

### ISSUE THREAD

**Issue by sk-cool** [hashicorp/terraform#6940]
Hi,

I am using terraform version v0.6.16
affected resource : aws_instance

The terraform plan fails with the following errors if I have more than 2 aws_instances in one aws_instances.tf.json file. The errors are as follows - 

_Errors:
- 1 error(s) occurred:
- module root: 4 error(s) occurred:
- Unknown root level key: instancei0afea5fc
- Unknown root level key: subnet_id
- Unknown root level key: tags
- Unknown root level key: vpc_security_group_ids_

Please note : This does not happen when I have only one instance per json file.
Also, this happens only for the type aws_instance. 
The plan works fine for other type of resources if I have multiple of them in one json file.

So, the question is does terraform support to have more than one instance in a json file?

Following is the example of aws_instances.tf.json that I am trying with.

``` json
{
  "resource": {
    "aws_instance": {
      "instancei060415f0": {
        "ami": "ami-118eb321",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "r3.large",
        "monitoring": "false",
        "private_ip": "10.252.1.240",
        "provisioner": {
          "remote-exec": {
            "inline": [
              [
                "#!/bin/bash",
                "hostname qagprivate02.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo qagprivate02.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=qagprivate02.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
              ]
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Env": "AMZN02",
          "Name": "qagprivate02"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgAPPhostrules.id}",
          "${aws_security_group.sginstanceaccesstoDB.id}",
          "${aws_security_group.sgallhostrules.id}"
        ]
      },
      "instancei06ce95f0": {
        "ami": "ami-e5e2ddd5",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "m1.medium",
        "monitoring": "false",
        "private_ip": "10.252.1.198",
        "provisioner": {
          "remote-exec": {
            "inline": [
              [
                "#!/bin/bash",
                "hostname vscn02.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo vscn02.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=vscn02.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
              ]
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Env": "AMZN02",
          "Name": "vscn02"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgAPPhostrules.id}",
          "${aws_security_group.sgSCAN.id}",
          "${aws_security_group.sgallhostrules.id}"
        ]
      },
      "instancei0afea5fc": {
        "ami": "ami-4dbf9e7d",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "t2.micro",
        "monitoring": "false",
        "private_ip": "10.252.1.241",
        "provisioner": {
          "remote-exec": {
            "inline": [
              [
                "#!/bin/bash",
                "hostname scantarget1.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo scantarget1.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=scantarget1.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
              ]
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Name": "scantarget1"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgSCANtarget.id}"
        ]
      },
      "instancei0c0415fa": {
        "ami": "ami-118eb321",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "r3.large",
        "monitoring": "false",
        "private_ip": "10.252.1.227",
        "provisioner": {
          "remote-exec": {
            "inline": [
              [
                "#!/bin/bash",
                "hostname qagdelta01.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo qagdelta01.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=qagdelta01.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
              ]
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Env": "AMZN02",
          "Name": "qagdelta01"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgAPPhostrules.id}",
          "${aws_security_group.sginstanceaccesstoDB.id}",
          "${aws_security_group.sgallhostrules.id}"
        ]
      }
    }
  }
}
```

Thanks,
Satish

**Comment by sk-cool** [comment]
Hi,

Any thoughts here? Can you please take a look? I want to create 70 instances and I think this should be possible though a single file.

Regards,
-Satish

**Comment by optimisticanshul** [comment]
@sk-cool I will try to look into it today, but can't promise.

**Comment by josephholsten** [comment]
@sk-cool I've taken a bit to minimize the test case, looks like I can still reproduce with:

```json
{
  "resource": {
    "aws_instance": {
      "instancei06ce95f0": {
        "provisioner": {
          "remote-exec": {
            "inline": [
              []
            ]
          }
        }
      },
      "instancei0afea5fc": {
        "provisioner": {
          "remote-exec": {
            "inline": [
              [
              ]
            ]
          }
        },
        "vpc_security_group_ids": []
      },
      "instancei0c0415fa": {
        "provisioner": {
          "remote-exec": {
            "inline": [
              [
              ]
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgallhostrules.id}"
        ]
      }
    }
  }
}
```

**Comment by sk-cool** [comment]
Yes, if there are more than two instances in a single file, then it does
not work.

**Comment by josephholsten** [comment]
@sk-cool I think there probably should be a better error message, but I think the problem is that the `provisioner.remote-exec.inline` has an array within an array. Do you have any better luck with:

```json
{
  "resource": {
    "aws_instance": {
      "instancei060415f0": {
        "ami": "ami-118eb321",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "r3.large",
        "monitoring": "false",
        "private_ip": "10.252.1.240",
        "provisioner": {
          "remote-exec": {
            "inline": [
                "#!/bin/bash",
                "hostname qagprivate02.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo qagprivate02.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=qagprivate02.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Env": "AMZN02",
          "Name": "qagprivate02"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgAPPhostrules.id}",
          "${aws_security_group.sginstanceaccesstoDB.id}",
          "${aws_security_group.sgallhostrules.id}"
        ]
      },
      "instancei06ce95f0": {
        "ami": "ami-e5e2ddd5",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "m1.medium",
        "monitoring": "false",
        "private_ip": "10.252.1.198",
        "provisioner": {
          "remote-exec": {
            "inline": [
                "#!/bin/bash",
                "hostname vscn02.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo vscn02.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=vscn02.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Env": "AMZN02",
          "Name": "vscn02"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgAPPhostrules.id}",
          "${aws_security_group.sgSCAN.id}",
          "${aws_security_group.sgallhostrules.id}"
        ]
      },
      "instancei0afea5fc": {
        "ami": "ami-4dbf9e7d",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "t2.micro",
        "monitoring": "false",
        "private_ip": "10.252.1.241",
        "provisioner": {
          "remote-exec": {
            "inline": [
                "#!/bin/bash",
                "hostname scantarget1.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo scantarget1.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=scantarget1.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Name": "scantarget1"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgSCANtarget.id}"
        ]
      },
      "instancei0c0415fa": {
        "ami": "ami-118eb321",
        "disable_api_termination": "false",
        "instance_initiated_shutdown_behavior": "stop",
        "instance_type": "r3.large",
        "monitoring": "false",
        "private_ip": "10.252.1.227",
        "provisioner": {
          "remote-exec": {
            "inline": [
                "#!/bin/bash",
                "hostname qagdelta01.${var.domain}",
                "if [ facter operatingsystemrelease | grep '^7' >/dev/null 2>&1 && echo 1 ]",
                "then",
                " echo qagdelta01.${var.domain} > /etc/hostname",
                "else",
                " sed -i 's/HOSTNAME=/HOSTNAME=qagdelta01.${var.domain}/' /etc/sysconfig/network-scripts/ifcfg-eth0",
                "fi"
            ]
          }
        },
        "subnet_id": "${aws_subnet.subnetad7efbda.id}",
        "tags": {
          "Env": "AMZN02",
          "Name": "qagdelta01"
        },
        "vpc_security_group_ids": [
          "${aws_security_group.sgAPPhostrules.id}",
          "${aws_security_group.sginstanceaccesstoDB.id}",
          "${aws_security_group.sgallhostrules.id}"
        ]
      }
    }
  }
}
```

**Comment by mitchellh** [comment]
This is a JSON parsing issue that as fixed in 0.7.7 I believe (or maybe 0.7.6).
